src/core/ledger/ledger.js

```diff
@@ -575,6 +602,52 @@ export class Ledger {
     account.identity = updatedIdentity;
   }
 
+  /**
+   * setPayoutAddress allows participants to set a payable address.
+   * These addresses are keyed on a specific currency, which ensures that
+   * users don't erroneously receive a grain distribution to an address that
+   * cannot handle it (such as a custodial wallet, or rigidly-designed
+   * contract) and effectively lose that reward.
+   *
+   * An address may be deleted by passing in an empty string for the
+   * `address` parameter. This is useful in case the underlying private key
+   * is compromised or the exchange hosting a custodial account is hacked.
+   */
+  setPayoutAddress(
+    id: IdentityId,
+    currency: CurrencyAddress,
+    address: EthAddress | ""
+  ): Ledger {
+    this._createAndProcessEvent({
+      type: "SET_PAYOUT_ADDRESS",
+      id,
+      currency,
+      address,
+    });
+    return this;
+  }
+  _setPayoutAddress({id, currency, address}: SetPayoutAddress) {
+    if (!this._accounts.has(id)) {
+      throw new Error(`setPayoutAddress: no identity matches id ${id}`);
+    }
+    const account = this._mutableAccount(id);
+    const currencyResult = ethAddressParser.parse(currency);
+    if (!currencyResult.ok) {
+      throw new Error(
+        `setPayoutAddress: invalid currency address: ${currency}`
+      );
+    }
+    if (address === "") {
+      account.addresses.delete(currency);
+      return;
+    }
+    const addressResult = ethAddressParser.parse(address);
+    if (!addressResult.ok) {
+      throw new Error(`setPayoutAddress: invalid payout address: ${address}`);
+    }
```

Comment:
having a method throw an error after a mutation is scary in this module, since it may leave the ledger in an inconsistent state. please follow the convention from the other methods, and have a clear demarcation between the argument-checking logic and error handling (above the demarcation line) and any mutation (below the demarcation line).

(in this case, since the mutation that is above the error checking returns immediately, it's safe, but it's not obviously safe, and it's easy to imagine someone in the future making an innocent-looking change that actually makes the code unsafe.)